### ISSUE TITLE
Rewrite-rules in Application Gateway cannot be set to only rewrite URL path and not query string

### ISSUE THREAD

**Issue by le-neumann** [hashicorp/terraform-provider-azurerm#14221]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v1.0.11
on darwin_amd64
+ provider registry.terraform.io/hashicorp/azurerm v2.85.0

### Affected Resource(s)

* `azurerm_application_gateway`

### Terraform Configuration Files

```hcl
rewrite_rule {
  name          = "path-rewrite"
  rule_sequence = 100
  url {
    path    = "/{var_uri_path_1}"
    reroute = false
  }
}
```

### Debug Output

[gist](https://gist.github.com/le-neumann/d575905e5c6b28308c5601889cbb7a42)

### Expected Behaviour

Since I did not enter a value for **query_string** in the **url** the expected behavior would be a rewrite-rule created that only rewrites the URL path. It should leave the query string as it is.

### Actual Behaviour

Terraform creates a rewrite rule with the "Components" set to "Both URL path and URL query string". Since there is no query string value set, it effectively deletes it. 

### Steps to Reproduce

1. terraform apply an application gateway resource with the above-mentioned rewrite-rule config.
3. Find the "Both URL path and URL query string" set when you check for the "Components" value in the UI.

### Important Factoids

It should be mentioned that the Azure CLI does not have the option to explicitly set the "components" value like it is possible in the web console. That being said, the CLI does set the value correctly according to the values **--modified-path** and **--modified-query-string**

**Comment by aristosvo** [comment]
Hi @le-neumann 👋 

This is a duplicate of a few other issues (will link them later), and potentially solved in PR #13899.

To work around this you can use `${var_query_?}`

**Comment by le-neumann** [comment]
Thanks @aristosvo this helped me a lot! 👍
